src/dryad/format/endf/createInteractionType.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef NJOY_DRYAD_FORMAT_ENDF_CREATEINTERACTIONTYPE
+#define NJOY_DRYAD_FORMAT_ENDF_CREATEINTERACTIONTYPE
+
+// system includes
+#include <vector>
+
+// other includes
+#include "tools/Log.hpp"
+#include "dryad/InteractionType.hpp"
+
+namespace njoy {
+namespace dryad {
+namespace format {
+namespace endf {
+
+  /**
+   *  @brief Create the interaction type based of the sublibrary value
+   */
+  InteractionType createInteractionType( int sublibrary ) {
+
+    switch ( sublibrary ) {
+
+      case     0 : return InteractionType::Nuclear;
+      case     3 : return InteractionType::Atomic;
+      case    10 : return InteractionType::Nuclear;
+      case   113 : return InteractionType::Atomic;
+      case 10010 : return InteractionType::Nuclear;
+      case 10020 : return InteractionType::Nuclear;
+      case 10030 : return InteractionType::Nuclear;
+      case 20030 : return InteractionType::Nuclear;
+      case 20040 : return InteractionType::Nuclear;
+      default : {
+
+        Log::error( "ENDF sublibrary {} does not define projectile-target data",
+                    sublibrary );
+        throw std::exception();
```

Comment:
Design question:
There exists libraries for heavy ions. They are almost invariably nuclear. Should the default be nuclear?
I don't know if those files' sublibraries are tagged that way though.

Reply:
I prefer throwing an exception to properly detect new sublibrary types we would have to support.

Reply:
Fair enough. I was going to comment that it would be adding an arbitrary restriction, but then I thought through the chain and realized that heavy ion ACE files would be a nightmare to finalize support.